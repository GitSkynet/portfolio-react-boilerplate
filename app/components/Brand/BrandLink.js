import styled from 'styled-components';
import PropTypes from 'prop-types';

const BrandLink = styled.a.attrs({
  href: ({ href }) => href || '/',
  title: ({ title }) => title || '← Back to the homepage',
})`
  display: block;
  height: 2.5rem;
  width: 2.5rem;
  color: #fff;
  overflow: hidden;
  text-decoration: none;

  @media (min-width: ${({ showNameBreakpoint }) => showNameBreakpoint}) {
    width: 100%;
  }
`;

BrandLink.propTypes = {
  href: PropTypes.string,
  showNameBreakpoint: PropTypes.string,
  title: PropTypes.string,
};

export default BrandLink;
