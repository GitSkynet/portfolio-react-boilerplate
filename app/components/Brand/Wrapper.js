import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 0.5rem;
  padding-bottom: 0;
`;

export default Wrapper;
