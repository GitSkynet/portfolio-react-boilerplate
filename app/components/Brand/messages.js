/*
 * BrandLink Messages
 *
 * This contains all the text for the BrandLink component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'portfolio.component.BrandLink.title',
    defaultMessage: '← Back to the homepage',
  },
});
