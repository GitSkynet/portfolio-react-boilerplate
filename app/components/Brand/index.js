import React from 'react';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';
import BrandLink from './BrandLink';
import Media from '../Media/index';
import MyLogo from '../../images/my-logo.svg';

const Brand = ({ href, imageSource, text }) => (
  <Wrapper>
    <BrandLink href={href}>
      <Media
        bodyAlign="middle"
        imageSource={imageSource}
        imageAlt="logo"
        imageAlign="middle"
        imageHeight="45px"
        imageWidth="45px"
      >{text}</Media>
    </BrandLink>
  </Wrapper>
);

Brand.propTypes = {
  href: PropTypes.string,
  imageSource: PropTypes.string,
  text: PropTypes.string,
};

Brand.defaultProps = {
  href: '/',
  imageSource: MyLogo,
  text: 'JOHAN MEESTER',
};

export default Brand;
