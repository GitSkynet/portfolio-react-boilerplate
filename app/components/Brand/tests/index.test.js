import React from 'react';
import { shallow } from 'enzyme';

import Brand from '../index';

describe('Brand', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Brand />);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
