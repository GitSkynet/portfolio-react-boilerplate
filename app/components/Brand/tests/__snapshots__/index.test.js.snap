// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Brand
    href="/"
    imageSource="IMAGE_MOCK"
    text="JOHAN MEESTER"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BrandLink__BrandLink
        href="/"
>
        <Media
                bodyAlign="middle"
                imageAlign="middle"
                imageAlt="logo"
                imageHeight="45px"
                imageSource="IMAGE_MOCK"
                imageWidth="45px"
                spacing="0"
        >
                JOHAN MEESTER
        </Media>
</BrandLink__BrandLink>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Media
          bodyAlign="middle"
          imageAlign="middle"
          imageAlt="logo"
          imageHeight="45px"
          imageSource="IMAGE_MOCK"
          imageWidth="45px"
          spacing="0"
>
          JOHAN MEESTER
</Media>,
        "href": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bodyAlign": "middle",
          "children": "JOHAN MEESTER",
          "imageAlign": "middle",
          "imageAlt": "logo",
          "imageHeight": "45px",
          "imageSource": "IMAGE_MOCK",
          "imageWidth": "45px",
          "spacing": "0",
        },
        "ref": null,
        "rendered": "JOHAN MEESTER",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrandLink__BrandLink
          href="/"
>
          <Media
                    bodyAlign="middle"
                    imageAlign="middle"
                    imageAlt="logo"
                    imageHeight="45px"
                    imageSource="IMAGE_MOCK"
                    imageWidth="45px"
                    spacing="0"
          >
                    JOHAN MEESTER
          </Media>
</BrandLink__BrandLink>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Media
            bodyAlign="middle"
            imageAlign="middle"
            imageAlt="logo"
            imageHeight="45px"
            imageSource="IMAGE_MOCK"
            imageWidth="45px"
            spacing="0"
>
            JOHAN MEESTER
</Media>,
          "href": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bodyAlign": "middle",
            "children": "JOHAN MEESTER",
            "imageAlign": "middle",
            "imageAlt": "logo",
            "imageHeight": "45px",
            "imageSource": "IMAGE_MOCK",
            "imageWidth": "45px",
            "spacing": "0",
          },
          "ref": null,
          "rendered": "JOHAN MEESTER",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
